### PR TITLE
feat(crons): Make a project level `MonitorCheckInIndex` endpoint

### PR DESCRIPTION
**Pull request by wedamija**
We want to make cron slugs unique per project instead of org. To support this, we need project level apis that the frontend can call.

Related to #65898

### CODE REVIEW

**Comment by wedamija** [comment]
This pr looks huge - it's just moving code around. I recommend reviewing with `git diff --color-moved=dimmed_zebra HEAD^`

File: src/sentry/api/base.py [hunk 1]
```diff
@@ -179,6 +179,23 @@ def create_audit_entry(self, request: Request, transaction_id=None, **kwargs):
...
+        paginator_cls=Paginator,
+        default_per_page: int | None = None,
+        max_per_page: int | None = None,
+        cursor_cls=Cursor,
+        response_cls=Response,
+        response_kwargs=None,
+        count_hits=None,
+        **paginator_kwargs,
+    ):
+        pass
```
Comment (cmanallen) [inline src/sentry/api/base.py:197]: Should we raise `NotImplemented`?

**Review by cmanallen — APPROVED** [review]
You're a machine Dan.

File: src/sentry/api/base.py (hunk 1 above)
Comment (wedamija) [reply src/sentry/api/base.py:197]: No need, the `abc.abstractmethod` takes care of that for us
```
>>> BaseEndpointMixin()
Traceback (most recent call last):
  File "/Users/wedamija/.pyenv/versions/3.11.6/lib/python3.11/code.py", line 90, in runcode
    exec(code, self.locals)
  File "<console>", line 1, in <module>
TypeError: Can't instantiate abstract class BaseEndpointMixin with abstract methods create_audit_entry, paginate, respond
```

**Pull request merged**